src/bitcoin-cli.cpp

```diff
@@ -291,6 +293,184 @@ class GetinfoRequestHandler: public BaseRequestHandler
+
+                is_asmap_on |= (mapped_as != 0);
+                max_peer_id_length = std::max(int(ToString(peer_id).length()), max_peer_id_length);
+                max_version_length = std::max(int((ToString(version) + sub_version).length()), max_version_length);
+            }
+        }
+        // Generate reports.
+        const UniValue& networkinfo{batch[ID_NETWORKINFO]["result"]};
+        std::string result{strprintf("%s %s - %i%s\n\n", PACKAGE_NAME, FormatFullVersion(), networkinfo["protocolversion"].get_int(), networkinfo["subversion"].get_str())};
+
+        // Report detailed peer connections list sorted by direction and minimum ping time.
+        if (m_verbose) {
+            std::sort(peers.begin(), peers.end());
+            result += "Peer connections sorted by direction and min ping\n<-> relay  conn minping   ping lastsend lastrecv uptime ";
+            if (is_asmap_on) result += " asmap ";
+            result += strprintf("%*s %-*s address\n", max_peer_id_length, "id", max_version_length, "version");
+            for (const m_peer& peer : peers) {
+                result += strprintf(
+                    "%3s %5s %5s%8d%7d %8s %8s%7s%*i %*s %-*s %s\n",
+                    peer.is_outbound ? "out" : "in",
+                    peer.is_block_relay ? "block" : "full",
+                    ConnTypeEnumToString(peer.conn_type),
+                    round(1000 * peer.min_ping),
+                    round(1000 * peer.ping),
+                    peer.last_send == 0 ? "" : ToString(time_now - peer.last_send),
+                    peer.last_recv == 0 ? "" : ToString(time_now - peer.last_recv),
+                    peer.conn_time == 0 ? "" : ToString((time_now - peer.conn_time) / 60),
+                    is_asmap_on ? 7 : 0, // variable spacing
+                    is_asmap_on && peer.mapped_as != 0 ? ToString(peer.mapped_as) : "",
+                    max_peer_id_length, // variable spacing
+                    peer.id,
+                    max_version_length, // variable spacing
+                    ToString(peer.version) + peer.sub_version,
+                    peer.addr);
+            }
+            result += "                     ms     ms      sec      sec    min\n\n";
+        }
+
+        // Report peer connection totals by type.
+        total_i = ipv4_i + ipv6_i + onion_i;
+        total_o = ipv4_o + ipv6_o + onion_o;
+        result += "Inbound and outbound peer connections\n";
+        result += strprintf("in:  ipv4 %3i  |  ipv6 %3i  |  onion %3i  |  total %3i  (%i block-relay)\n", ipv4_i, ipv6_i, onion_i, total_i, block_relay_i);
+        result += strprintf("out: ipv4 %3i  |  ipv6 %3i  |  onion %3i  |  total %3i  (%i block-relay)\n", ipv4_o, ipv6_o, onion_o, total_o, block_relay_o);
+        result += strprintf("all: %i\n", total_i + total_o);
+
+        // Report local addresses, ports, and scores.
+        result += "\nLocal addresses";
```

Comment:
Given that the `Local addresses` header may be followed by nothing, what about skipping it if there are no local addresses or printing something like `none` or `n/a` below it in that case?

Reply:
Skipping it might violate the principle of least surprise. I've never seen it followed by nothing, though.

Reply:
I see this:
```
$ ./src/bitcoin-cli -netinfo 
Bitcoin Core v0.20.99.0-f63ceccca - 70015/Satoshi:0.20.0/

Inbound and outbound peer connections
in:  ...
out: ...
all: ...

Local addresses
$ 
```

I guess it happens when the node does not have an incoming connections and `--externalip=` is not provided to `bitcoind`.

Reply:
Same here, behind a firewall with no port forwarding, my local addresses is empty. Not sure the best way to handle it visually. Perhaps even just an empty newline.

Reply:
Can propose adding "n/a\n" in the follow-up.